NETProvider/src/FirebirdSql.Data.FirebirdClient/Client/Managed/Version10/SrpClient.cs

```diff
@@ -0,0 +1,220 @@
+﻿/*
+ *	Firebird ADO.NET Data provider for .NET and Mono
+ *
+ *	   The contents of this file are subject to the Initial
+ *	   Developer's Public License Version 1.0 (the "License");
+ *	   you may not use this file except in compliance with the
+ *	   License. You may obtain a copy of the License at
+ *	   http://www.firebirdsql.org/index.php?op=doc&id=idpl
+ *
+ *	   Software distributed under the License is distributed on
+ *	   an "AS IS" basis, WITHOUT WARRANTY OF ANY KIND, either
+ *	   express or implied. See the License for the specific
+ *	   language governing rights and limitations under the License.
+ *
+ *	Copyright (c) 2015 Hajime Nakagami<nakagami@gmail.com>
+ *	All Rights Reserved.
+ *
+ */
+
+// http://srp.stanford.edu/design.html
+// https://github.com/FirebirdSQL/jaybird/blob/master/src/main/org/firebirdsql/gds/ng/wire/SrpClient.java
+
+using System;
+using System.Linq;
+using System.Collections.Generic;
+using System.Numerics;
+using System.Security.Cryptography;
+using System.Globalization;
+
+namespace FirebirdSql.Data.Client.Managed.Version10
+{
+	public class SrpClient {
+		private const int SRP_KEY_SIZE = 128;
+		private const int SRP_SALT_SIZE = 32;
+		private static BigInteger N = BigInteger.Parse("00E67D2E994B2F900C3F41F08F5BB2627ED0D49EE1FE767A52EFCD565CD6E768812C3E1E9CE8F0A8BEA6CB13CD29DDEBF7A96D4A93B55D488DF099A15C89DCB0640738EB2CBDD9A8F7BAB561AB1B0DC1C6CDABF303264A08D1BCA932D1F1EE428B619D970F342ABA9A65793B8B2F041AE5364350C16F735F56ECBCA87BD57B29E7", NumberStyles.HexNumber);
+		private static BigInteger g = new BigInteger(2);
+		private static BigInteger k = BigInteger.Parse("1277432915985975349439481660349303019122249719989");
+
+		private static byte[] SEPARATOR_BYTES = System.Text.Encoding.UTF8.GetBytes(":");
+
+		private BigInteger publicKey;	/* A */
+		private BigInteger privateKey;	/* a */
+		private byte[] proof;			/* M */
+		private byte[] sessionKey;		/* K */
+
+		private static BigInteger fromBigByteArray(byte [] b) {
+			Array.Reverse(b);
+			return new BigInteger(b.Concat(new byte[] { 0 }).ToArray());
```

Comment:
If first byte larger than 128 converted as negative integer.
This situation Big number needs positive.